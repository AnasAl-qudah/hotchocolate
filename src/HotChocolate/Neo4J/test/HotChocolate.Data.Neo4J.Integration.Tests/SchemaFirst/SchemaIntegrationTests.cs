using CookieCrumble;
using HotChocolate.Data.Neo4J.Testing;
using HotChocolate.Execution;

namespace HotChocolate.Data.Neo4J.Integration.Tests.SchemaFirst;

[Collection(Neo4JDatabaseCollectionFixture.DefinitionName)]
public class SchemaIntegrationTests : IClassFixture<Neo4JFixture>
{
    private readonly Neo4JDatabase _database;
    private readonly Neo4JFixture _fixture;

    public SchemaIntegrationTests(Neo4JDatabase database, Neo4JFixture fixture)
    {
        _database = database;
        _fixture = fixture;
    }

    [Fact]
    public async Task MoviesSchemaIntegrationTests_GetSchema()
    {
        // arrange
        var tester = await _fixture.Arrange(_database);

        await Snapshot.Create()
            .Add(tester.Schema, "Schema")
            .MatchAsync();
    }

    [Fact]
    public async Task MoviesSchemaIntegrationTests()
    {
        // arrange
        var tester = await _fixture.Arrange(_database);

        // act
        var res1 = await tester.ExecuteAsync(
            @"{
                actors {
                    name
                    actedIn {
                        title
                    }
                }
            }");

        var res2 = await tester.ExecuteAsync(
            @"{
                actors (where : {name : { startsWith : ""Keanu"" }}) {
                    name
                    actedIn {
                        title
                    }
                }
            }");

        var res3 = await tester.ExecuteAsync(
            @"{
                movies {
                    title
                }
            }");

        var res4 = await tester.ExecuteAsync(
            @"{
                actors(order: [{ name : ASC }]) {
                    name
                    actedIn {
                        title
                    }
                }
            }");

        var res5 = await tester.ExecuteAsync(@"{
                actors(order: [{ name : ASC }] skip: 1 take: 2) {
                    items {
                        name
                        actedIn {
                            title
                        }
                    }
                }
            }");

        // assert
        await Snapshot.Create()
            .AddResult(res1, "MoviesSchema_Actors_Query")
            .AddResult(res2, "MoviesSchema_Name_StartsWith_Actors_Query")
            .AddResult(res3, "MoviesSchema_Movies_Query")
            .AddResult(res4, "MoviesSchema_Name_Desc_Sort_Actors_Query")
            .AddResult(res5, "MoviesSchema_Name_Desc_Sort_Actors_SkipOne_TakeTwo_Query")
            .MatchAsync();
    }
}
