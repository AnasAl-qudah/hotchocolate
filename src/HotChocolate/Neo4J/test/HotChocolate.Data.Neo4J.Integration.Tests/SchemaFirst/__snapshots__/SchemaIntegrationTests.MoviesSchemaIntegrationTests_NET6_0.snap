MoviesSchema_Actors_Query Result:
---------------
{
  "errors": [
    {
      "message": "The field \u0060name\u0060 does not exist on the type \u0060ActorsCollectionSegment\u0060.",
      "locations": [
        {
          "line": 3,
          "column": 21
        }
      ],
      "path": [
        "actors"
      ],
      "extensions": {
        "type": "ActorsCollectionSegment",
        "field": "name",
        "responseName": "name",
        "specifiedBy": "http://spec.graphql.org/October2021/#sec-Field-Selections-on-Objects-Interfaces-and-Unions-Types"
      }
    },
    {
      "message": "The field \u0060actedIn\u0060 does not exist on the type \u0060ActorsCollectionSegment\u0060.",
      "locations": [
        {
          "line": 4,
          "column": 21
        }
      ],
      "path": [
        "actors"
      ],
      "extensions": {
        "type": "ActorsCollectionSegment",
        "field": "actedIn",
        "responseName": "actedIn",
        "specifiedBy": "http://spec.graphql.org/October2021/#sec-Field-Selections-on-Objects-Interfaces-and-Unions-Types"
      }
    }
  ]
}
---------------

MoviesSchema_Name_StartsWith_Actors_Query Result:
---------------
{
  "errors": [
    {
      "message": "The field \u0060name\u0060 does not exist on the type \u0060ActorsCollectionSegment\u0060.",
      "locations": [
        {
          "line": 3,
          "column": 21
        }
      ],
      "path": [
        "actors"
      ],
      "extensions": {
        "type": "ActorsCollectionSegment",
        "field": "name",
        "responseName": "name",
        "specifiedBy": "http://spec.graphql.org/October2021/#sec-Field-Selections-on-Objects-Interfaces-and-Unions-Types"
      }
    },
    {
      "message": "The field \u0060actedIn\u0060 does not exist on the type \u0060ActorsCollectionSegment\u0060.",
      "locations": [
        {
          "line": 4,
          "column": 21
        }
      ],
      "path": [
        "actors"
      ],
      "extensions": {
        "type": "ActorsCollectionSegment",
        "field": "actedIn",
        "responseName": "actedIn",
        "specifiedBy": "http://spec.graphql.org/October2021/#sec-Field-Selections-on-Objects-Interfaces-and-Unions-Types"
      }
    }
  ]
}
---------------

MoviesSchema_Movies_Query Result:
---------------
{
  "errors": [
    {
      "message": "The field \u0060title\u0060 does not exist on the type \u0060MoviesCollectionSegment\u0060.",
      "locations": [
        {
          "line": 3,
          "column": 21
        }
      ],
      "path": [
        "movies"
      ],
      "extensions": {
        "type": "MoviesCollectionSegment",
        "field": "title",
        "responseName": "title",
        "specifiedBy": "http://spec.graphql.org/October2021/#sec-Field-Selections-on-Objects-Interfaces-and-Unions-Types"
      }
    }
  ]
}
---------------

MoviesSchema_Name_Desc_Sort_Actors_Query Result:
---------------
{
  "errors": [
    {
      "message": "The field \u0060name\u0060 does not exist on the type \u0060ActorsCollectionSegment\u0060.",
      "locations": [
        {
          "line": 3,
          "column": 21
        }
      ],
      "path": [
        "actors"
      ],
      "extensions": {
        "type": "ActorsCollectionSegment",
        "field": "name",
        "responseName": "name",
        "specifiedBy": "http://spec.graphql.org/October2021/#sec-Field-Selections-on-Objects-Interfaces-and-Unions-Types"
      }
    },
    {
      "message": "The field \u0060actedIn\u0060 does not exist on the type \u0060ActorsCollectionSegment\u0060.",
      "locations": [
        {
          "line": 4,
          "column": 21
        }
      ],
      "path": [
        "actors"
      ],
      "extensions": {
        "type": "ActorsCollectionSegment",
        "field": "actedIn",
        "responseName": "actedIn",
        "specifiedBy": "http://spec.graphql.org/October2021/#sec-Field-Selections-on-Objects-Interfaces-and-Unions-Types"
      }
    }
  ]
}
---------------

MoviesSchema_Name_Desc_Sort_Actors_SkipOne_TakeTwo_Query
---------------
{
  "errors": [
    {
      "message": "Cannot handle the specified data source."
    }
  ],
  "data": {
    "actors": null
  }
}
---------------
